### DISCUSSION TITLE
PDF generation using an existing Form

### DISCUSSION THREAD

**Discussion by prateekjsingh** [formio/formio.js#5070] [Q&A]
Hi FormIO team,
I am using FormIO library's minified version to embed the FormBuilder into my application.
The ask is to generate and save a PDF version of the Form at the time of submit.

From the documentation it seems like we can use the PDF basic server for free and I need to get a license for PDF server plus.

Can you please confirm which API method can be used to generate the FORM pdf at the time of submission?

I tried using the getDownloadUrl method of Formio class and it returns a blank url.

Thanks
Prateek Jeet Singh

**Comment by brendanbond** [comment]
Hi, @prateekjsingh 

> From the documentation it seems like we can use the PDF basic server for free and I need to get a license for PDF server plus.

I believe that this is a misunderstanding. Our "PDF Basic" server, and the ability to generate PDFs from form submissions, is available via our enterprise offering, for which you can find pricing information [here](https://form.io/pricing).
